### PR TITLE
chore: major cleanup of the code and build

### PR DESCRIPTION
**Pull request by smira**
There should no functional changes, code was put through `Kres` for
automated builds.

Go bumped, protobuf & grpc libraries bumped, linting errors fixed, etc.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

File: .markdownlint.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# THIS FILE WAS AUTOMATICALLY GENERATED, PLEASE DO NOT EDIT.
```
Comment (Unix4ever) [inline .markdownlint.json:1]: Not related to this PR, but I wonder why `markdownlint` is fine having these `#` comments here :)
This doesn't look like a valid json.

**Review by Unix4ever — APPROVED** [review]

File: .markdownlint.json (hunk 1 above)
Comment (smira) [reply .markdownlint.json:1]: I believe `#` is a valid comment in JSON

Comment (Unix4ever) [reply .markdownlint.json:1]: yeah, probably valid for markdown lint :) But it doesn't look to be a standard.

**Pull request merged**